runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -172,48 +176,53 @@ public DirectPipelineResult run(Pipeline originalPipeline) {
     }
     pipeline.replaceAll(defaultTransformOverrides());
     MetricsEnvironment.setMetricsSupported(true);
```

Comment:
This probably should restore the previous state, whatever it happened to be, rather than blind-setting to false; However, I'm not super worried about it, because this is already global state so multiple pipelines can interfere with each other.